Analysis/OverlayRemoval/src/CheatedMCOverlayRemoval.cc

```diff
@@ -0,0 +1,344 @@
+				 _MCParticleCollectionName ,
+				 std::string("MCParticlesSkimmed")
+				 );
+
+	registerInputCollection( LCIO::RECONSTRUCTEDPARTICLE,
+				 "RecoParticleCollection" ,
+				 "Name of the ReconstructedParticles input collection"  ,
+				 _recoParticleCollectionName ,
+				 std::string("PandoraPFOs")
+				 );
+
+	registerInputCollection( LCIO::LCRELATION,
+				 "RecoMCTruthLink",
+				 "Name of the RecoMCTruthLink input collection"  ,
+				 _recoMCTruthLink,
+				 std::string("RecoMCTruthLink")
+				 );
+
+	registerInputCollection( LCIO::LCRELATION,
+                                 "MCTruthRecoLink",
+                                 "Name of the MCTruthRecoLink input collection"  ,
+                                 _mcTruthRecoLink,
+                                 std::string("MCTruthRecoLink")
+                                 );
+
+	registerOutputCollection(LCIO::RECONSTRUCTEDPARTICLE,
+				 "OutputPfoCollection",
+				 "Name of output PFO collection",
+				 _OutputPfoCollection,
+				 std::string("PFOsWithoutOverlay")
+				 );	
+	registerOutputCollection(LCIO::RECONSTRUCTEDPARTICLE,
+				 "OutputOverlayCollection",
+				 "Name of output Overlay collection",
+				 _OutputOverlayCollection,
+				 std::string("PFOsFromOverlay")
+				 );
+}
+
+void CheatedMCOverlayRemoval::init()
+{
+
+	streamlog_out(DEBUG) << "   init called  " << std::endl ;
+	this->Clear();
+
+}
+
+void CheatedMCOverlayRemoval::Clear()
+{
+  m_nAllPFOs = 0;
+  m_nMCPs = 0;
+}
+
+void CheatedMCOverlayRemoval::processRunHeader()
+{
+  streamlog_out(DEBUG0) << "   processRunHeader called" << std::endl ;
+  m_nRun++ ;  
+  streamlog_out(DEBUG0) << "   processRunHeader finished successfully" << std::endl ;
+}
+
+void CheatedMCOverlayRemoval::processEvent( LCEvent *pLCEvent )
+{
+  this->Clear();
+  m_nRun = pLCEvent->getRunNumber();
+  m_nEvt = pLCEvent->getEventNumber();
+  streamlog_out(DEBUG) << "Processing event " << pLCEvent->getEventNumber() << std::endl;
+
+  const EVENT::LCCollection *PFOs{};
+  const EVENT::LCCollection *MCPs{};
+  IMPL::LCCollectionVec* OutputPfoCollection(NULL);
+  OutputPfoCollection = new IMPL::LCCollectionVec(LCIO::RECONSTRUCTEDPARTICLE);
+  OutputPfoCollection->setSubset( true ); 
+  IMPL::LCCollectionVec* OutputOverlayCollection(NULL);
+  OutputOverlayCollection = new IMPL::LCCollectionVec(LCIO::RECONSTRUCTEDPARTICLE);
+  OutputOverlayCollection->setSubset( true ); 
+  try
+    {
+      PFOs = pLCEvent->getCollection( _recoParticleCollectionName );
+      MCPs = pLCEvent->getCollection( _MCParticleCollectionName );
+
+      LCRelationNavigator RecoMCParticleNav( pLCEvent->getCollection( _recoMCTruthLink ) );
+      LCRelationNavigator MCParticleRecoNav( pLCEvent->getCollection( _mcTruthRecoLink ) );
+
+      m_nAllPFOs = PFOs->getNumberOfElements();
+      m_nMCPs = MCPs->getNumberOfElements();
+      int nKeptPFOs = 0;
+      int nRemovedPFOs = 0;
+      
+      //Create new collection without PFOs from MCPs that are overlay
```

Comment:
```suggestion
      //Create new collection with PFOs from MCPs that are overlay
```
At least this is what seems to be done in this loop.